### PR TITLE
594 bug fix baseline johntheripper

### PR DESCRIPTION
**Pull request by slpysiren**
Completed the baseline checklist for the component. Please let me know if I missed anything or need to make any changes.

### CODE REVIEW

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:42]: These are all react state hooks as indicated by the "useState". Please move them into that appropriate section. 
```suggestion
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:37]:
```suggestion
    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
    const [selectedFileTypeOption, setSelectedFileTypeOption] = useState(""); // State variable to store the selected file type.
    const [selectedModeOption, setSelectedModeOption] = useState(""); // State variable to store the selected crack mode.
    const [selectedIncrementOption, setSelectedIncrementOption] = useState(""); // State variable to store the selected increment order.

```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:48]: Remove this comment.

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:66]:
```suggestion
        "Step 2: Specify the hashing algorithm used by the password you are trying to crack (e.g md5).\n" +
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:67]:
```suggestion
        "Step 3: Specify the cracking mode to use.\n" +
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:68]:
```suggestion
        "Step 4: Select the file type.\n" +
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:70]:
```suggestion
        "Step 6: Click 'Start John the Ripper'.\n" +
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:71]:
```suggestion
        "Step 7: View the output block to view the results of the tools execution.\n";
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:209]:
```suggestion
            // Crack the extracted hash
```

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (bradleybogg) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:217]: Update this block to use the overlay for cancelling

**Review by bradleybogg — CHANGES_REQUESTED** [review]
Very good work @slpysiren

I appreciate your efforts in doing it right and taking the time to do so.
Good work on this one. I have some very minor changes to make. Some of the comments for steps I've added reccomendations. See which you think is better. For the others please implement changes.

Thank you!

**Review by michaeljpigott — APPROVED** [review]
Hi @slpysiren 

This baselining work looks good. I approve to merge pending app testing from @zkkok 

Kind regards

Michael

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (zkkok) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:1]: Remove unused imports

File: src/components/JohnTheRipper/JohnTheRipper.tsx
Comment (zkkok) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:14]: Revise comment

**Review by zkkok — CHANGES_REQUESTED** [review]
Hi @slpysiren, Application testing has been done. There's a few issues but most of them pre-existed within the component. Only need to make some minor changes before it can be merged.

Summary of functionality test:
[Functionlity Testing(John).pdf](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/15282610/Functionlity.Testing.John.pdf)

**Comment by michaeljpigott** [comment]
Hi @zkkok, I will let you review your requested changes for this one. 

Thanks.

**Review by zkkok — APPROVED** [review]
Baseline bug fix looks good without any issue in sight.

**Pull request merged**
